File: Excel/574 Challenge.py
import pandas as pd

path = "574 Sort Numbers in Odd Positions Only.xlsx"
input = pd.read_excel(path, usecols="A", nrows=10)
test = pd.read_excel(path, usecols="B", nrows=10, dtype=str)

def process_numbers(number):
    number = list(str(number))
    odd_indices = range(0, len(number), 2)
    odd_numbers = sorted(int(number[i]) for i in odd_indices)
    for i, idx in enumerate(odd_indices):
        number[idx] = str(odd_numbers[i])
    return ''.join(number)

input['Answer Expected'] = input.iloc[:, 0].apply(process_numbers)

print(test['Answer Expected'].equals(input['Answer Expected']))

File: Excel/500-599/574/574 Challenge.R
library(tidyverse)
library(readxl)

path = "Excel/574 Sort Numbers in Odd Positions Only.xlsx"
input = read_excel(path, range = "A1:A10")
test  = read_excel(path, range = "B1:B10")

process_numbers = function(number) {
  number = strsplit(as.character(number), "")[[1]]
  odd = seq(1, length(number), by = 2)
  number[odd] = sort(as.numeric(number[odd]))
  paste(number, collapse = "")
}

result = input %>%
  mutate(`Answer Expected` = map_chr(Numbers, process_numbers))

all.equal(result$`Answer Expected`, test$`Answer Expected`)
#> [1] TRUE